src/main/java/net/snowflake/ingest/streaming/internal/RowBufferStats.java

```diff
@@ -38,18 +41,29 @@ class RowBufferStats {
   /** Display name is required for the registration endpoint */
   private final String columnDisplayName;
 
+  private final boolean enableDistinctValue;
+  private Set<Object> distinctValues;
+
   /** Creates empty stats */
   RowBufferStats(
-      String columnDisplayName, String collationDefinitionString, int ordinal, Integer fieldId) {
+      String columnDisplayName,
+      String collationDefinitionString,
+      int ordinal,
+      Integer fieldId,
+      boolean enableDistinctValue) {
     this.columnDisplayName = columnDisplayName;
     this.collationDefinitionString = collationDefinitionString;
     this.ordinal = ordinal;
     this.fieldId = fieldId;
+    this.enableDistinctValue = enableDistinctValue;
+    if (enableDistinctValue) {
+      this.distinctValues = new HashSet<>();
```

Comment:
is there any concurrent access to this collection?

Reply:
The value parser parse every row in a chunk in order. There's no concurrent access within the same `RowBufferStats`